src/secret_sharing/xor.rs

```diff
@@ -19,4 +23,44 @@ impl XorReplicated {
     pub fn right(&self) -> u64 {
         self.right
     }
+
+    /// Deserializes this instance from a slice of bytes
+    ///
+    /// ## Errors
+    /// if buffer does not have enough capacity to hold a valid value of this instance.
+    #[allow(clippy::missing_panics_doc)]
+    pub fn deserialize(buf: &[u8]) -> io::Result<Self> {
+        if buf.len() < Self::SIZE_IN_BYTES {
+            Err(io::Error::new(
+                io::ErrorKind::UnexpectedEof,
+                "not enough buffer capacity",
+            ))
+        } else {
+            let left_buf: [u8; 8] = buf[..8].try_into().unwrap();
+            let right_buf: [u8; 8] = buf[8..16].try_into().unwrap();
+
+            Ok(Self {
+                left: u64::from_le_bytes(left_buf),
+                right: u64::from_le_bytes(right_buf),
+            })
+        }
+    }
+
+    /// Serializes this instance into a mutable slice of bytes, writing from index 0.
+    ///
+    /// ## Errors
+    /// if buffer capacity is not sufficient to hold all the bytes.
+    pub fn serialize(&self, buf: &mut [u8]) -> io::Result<usize> {
+        if buf.len() < Self::SIZE_IN_BYTES {
+            Err(io::Error::new(
+                io::ErrorKind::WriteZero,
+                "not enough buffer capacity",
+            ))
+        } else {
+            buf[..8].copy_from_slice(&self.left.to_le_bytes());
+            buf[8..16].copy_from_slice(&self.right.to_le_bytes());
```

Comment:
Is there any reason you can't:

```suggestion
            buf[8..].copy_from_slice(&self.right.to_le_bytes());
```
... on the expectation that you will only be given SIZE_IN_BYTES bytes to write to?

Reply:
yea the way it is written, it supports both cases (exactly 2*SIZE_IN_BYTES and more)